### ISSUE TITLE
ApiImplicitBody decorator Enum type support

### ISSUE THREAD

**Issue by BMalaichik** [nestjs/swagger#144]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Using `@ApiImplicitBody` decorator doesn't support enum-value definitions
Only `ApiModelProperty` supports it.

## Expected behavior
Use enum-value definition in `ApiImplicitBody` definitions

## What is the motivation / use case for changing the behavior?
I don't want to define extra Dto, if I use only single body property which is enum value.

Also, anothger thing: why not to get rid of current enum definitions, like:
<code>@ApiModelProperty({ enum:  [ 'value1', ''value2 ] })</code>
and under-the hood get enum values with `_.values(EnumSource)`, which produces enum array
<code>@ApiModelProperty({ enum:  { source: EnumSource  } })</code>
for me it a bit frustrating, duplicating enum values in api definitions...

## Environment

@nestjs/swagger@2.5.1

**Comment by nartc** [comment]
Regarding the issue, I’d play around with it to see if it makes sense. If the enum is the only argument of a route handler, I would use @ApiImplicitQuery 

You can use the enum directly when defining an ApiModelProperty like:

@ApiModelProperty({enum: UserRole})

**Comment by BMalaichik** [comment]
@nartc according to the doc, I must provide list of values for enum definition
https://docs.nestjs.com/recipes/swagger
`Working with enums` section

I am also not sure if it's really necessary, I wanna hear some other opinions regarding this.
anyway, major thing would be about avoiding values explicit enumeration

**Comment by nartc** [comment]
That’s my fault as when I created the docs for the Enum part, the SwaggerEnumType had just been put in. 

I can confirm that you can use your Enum definition to pass in for enum property for ApiModelProperty

I’ll make changes to the docs ASAP

**Comment by BMalaichik** [comment]
@nartc amazing, thanks!
anyway, let;s keep issue opened to hear another guys opinion regarding need of defining implicit body enum parameter

Thanks!

**Comment by zbron99** [comment]
@nartc @BMalaichik Agreed, please add to docs UI! I was so bummed out to see that the nicer implicit UI is not available for APIImplicitBody haha. I'll be updating as soon as I hear it's ready. Thank you!

**Comment by nartc** [comment]
@BMalaichik @zbron99 Can you please give me a use-case of Enum value in a `ApiImplicitBody` ?

**Comment by BMalaichik** [comment]
@nartc I just provide as part of body `types` enum values array

```
    @Put("/:id/data")
    @ApiImplicitBody({ name: "types", type: DataTypeEnum, isArray: true })
    public async updateData(
        @Param("id", new ParseIntPipe()) id: number,
        @Body("types") dataType: DataTypeEnum[],
    ): Promise<void> {
        await serviceCall...
    }
```

**Comment by nartc** [comment]
@BMalaichik Sure, I'll look into it more. Thanks!

**Comment by nartc** [comment]
@BMalaichik After looking around a bit more for example regarding Body parameter with Enum in Swagger, no luck :(

**Comment by kamilmysliwiec** [comment]
Fixed in 4.0.1
